iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalTableOperations.java

```diff
@@ -278,8 +278,11 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
             serializedSnapshotRefs == null
                 ? new HashMap<>()
                 : SnapshotsUtil.parseSnapshotRefs(serializedSnapshotRefs);
+
+        // Multi-branch support is now enabled with snapshot ID matching
+
```

Comment:
It looks like https://github.com/linkedin/openhouse/pull/379/files#diff-526cc78ff45a139113a29951f4db0a28242b9a7f3bc9caef9ceec26b04c5aeb8R271 also computes the diff between current and new, similar to newly added method applySnapshotOperations. Is there a way to consolidate or should they remain the same?